include/fmt/format-inl.h

```diff
@@ -939,7 +938,11 @@ template <int GRISU_VERSION> struct grisu_shortest_handler {
                           uint64_t error, int exp, bool integral) {
     buf[size++] = digit;
     if (remainder >= error) return digits::more;
+#ifdef __cpp_if_constexpr
+    if constexpr (GRISU_VERSION != 3) {
+#else
     if (GRISU_VERSION != 3) {
+#endif
```

Comment:
Here and below: I suggest using `const_check` from

https://github.com/fmtlib/fmt/blob/d39ebf3ff251bf199f88f917cecc504230823342/include/fmt/printf.h#L21

which was introduced specifically to silence this warning. It will avoid duplication.

E.g.

```
if (const_check(GRISU_VERSION != 3)) {
```

Reply:
Fixed